src/NSwag.CodeGeneration.CSharp.Tests/PlainResponseTests.cs

```diff
@@ -39,15 +103,59 @@ public async Task When_openapi3_reponse_contains_plain_text_then_Convert_is_gene
 }";
             var document = await OpenApiDocument.FromJsonAsync(json);
 
-            //// Act
+            // Act
             var codeGenerator = new CSharpClientGenerator(document, new CSharpClientGeneratorSettings
             {
                 GenerateClientInterfaces = true
             });
             var code = codeGenerator.GenerateFile();
 
-            //// Assert
+            // Assert
             Assert.Contains("public async System.Threading.Tasks.Task<string> PlainAsync(", code);
+            Assert.Contains("(string)System.Convert.ChangeType(responseData_, typeof(string));", code);
+        }
+
+        [Fact]
+        public async Task When_swagger_reponse_contains_plain_text_string_array_then_ReadObjectResponseAsync_is_generated()
+        {
+            // Arrange
+            var swagger = @"{
+  ""swagger"": ""2.0"",
+  ""paths"": {
+    ""/api/get-ienumerable"": {
+      ""get"": {
+        ""produces"": [
+          ""text/plain"",
+          ""application/json"",
+          ""text/json""
+        ],
+        ""responses"": {
+          ""200"": {
+            ""description"": ""Success"",
+            ""schema"": {
+              ""type"": ""array"",
```

Comment:
Can you add a test for having type as string instead of array? Because it doesn't work when the type is directly a string. The assertion then being `Assert.Contains("var responseData_ = response_.Content == null ? null : await response_.Content.ReadAsStringAsync().ConfigureAwait(false);")` and `Assert.Contains("var result_ = (string)System.Convert.ChangeType(responseData_, typeof(string));")`.

Reply:
Can you or someone create a PR for this?